lib/cext/preprocess.rb

```diff
@@ -145,6 +165,229 @@
+        replacement: 'wrapper->active_thread = rb_tr_handle_for_managed(thread_current);'
+      },
+      {
+        match: /wrapper->active_thread == thread_current/,
+        replacement: 'rb_tr_managed_from_handle(wrapper->active_thread) == thread_current'
+      },
+      {
+        match: /rb_inspect\(wrapper->active_thread\)/,
+        replacement: 'rb_inspect(rb_tr_managed_from_handle(wrapper->active_thread))'
+      },
+      {
+        match: /args.sql = rb_str_export_to_enc\(sql, rb_to_encoding\(wrapper->encoding\)\);/,
+        replacement: 'args.sql = rb_tr_handle_for_managed_leaking(rb_str_export_to_enc(sql, rb_to_encoding(rb_tr_managed_from_handle(wrapper->encoding)));'
+      },
+      {
+        match: /args.sql = sql/,
+        replacement: 'args.sql = rb_tr_handle_for_managed_leaking(sql)'
+      },
+      {
+        match: /args\.sql_ptr = RSTRING_PTR\(args\.sql\);/,
+        replacement: 'args.sql_ptr = RSTRING_PTR(rb_tr_managed_from_handle(args.sql));'
+      },
+      {
+        match: /args\.sql_len = RSTRING_LEN\(args\.sql\);/,
+        replacement: 'args.sql_len = RSTRING_LEN(rb_tr_managed_from_handle(args.sql));'
+      },
+      {
+        match: /async_args\.self = self;/,
+        replacement: 'async_args.self = rb_tr_handle_for_managed_leaking(self);'
+      },
+      {
+        match: /conn_enc = rb_to_encoding\(wrapper->encoding\);/,
+        replacement: 'conn_enc = rb_to_encoding(rb_tr_managed_from_handle(wrapper->encoding));'
+      },
+      {
+        match: /wrapper->encoding = rb_enc;/,
+        replacement: 'wrapper->encoding = rb_tr_handle_for_managed(rb_enc);'
+      }
+    ]
+  },
+  'result.c' => {
+    gem: 'mysql2',
+    patches: [
+      {
+        match: /wrapper->statement != Qnil/,
+        replacement: 'rb_tr_managed_from_handle(wrapper->statement) != Qnil'
+      },
+      {
+        match: /wrapper->client != Qnil/,
+        replacement: 'rb_tr_managed_from_handle(wrapper->client) != Qnil'
+      },
+      {
+        match: /xfree\(wrapper\);/,
+        replacement: MYSQL2_FREE_RESULT_WRAPPER
+      },
+      {
+        match: /wrapper->fields == Qnil/,
+        replacement: 'rb_tr_managed_from_handle(wrapper->fields) == Qnil'
+      },
+      {
+        match: /wrapper->fields = rb_ary_new2\(wrapper->numberOfFields\);/,
+        replacement: 'wrapper->fields = rb_tr_handle_for_managed(rb_ary_new2(wrapper->numberOfFields));'
+      },
+      {
+        match: /rb_ary_entry\(wrapper->fields,/,
+        replacement: 'rb_ary_entry(rb_tr_managed_from_handle(wrapper->fields),'
+      },
+      {
+        match: /rb_to_encoding\(wrapper->encoding\)/,
+        replacement: 'rb_to_encoding(rb_tr_managed_from_handle(wrapper->encoding))'
+      },
+      {
+        match: /rb_ary_store\(wrapper->fields,/,
+        replacement: 'rb_ary_store(rb_tr_managed_from_handle(wrapper->fields),'
+      },
+      {
+        match: /RARRAY_LEN\(wrapper->fields\)/,
```

Comment:
`/\bwrapper->fields\s*([\),;]|==)/` is probably good enough to match reads of a field.
And then something like
`/\bwrapper->fields\s*=\s*(\w.+);\s*$/` for writes to a field.

Reply:
I've updated to use these, thanks.